From de1608cda80c44aef86a6077067c48c5c0ef89fc Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:56 -0400
Subject: [ethernet] i40e: Add functionality for FD SB to drop packets

Message-id: <1409658042-22689-56-git-send-email-sassmann@redhat.com>
Patchwork-id: 90250
O-Subject: [RHEL7.1 PATCH 055/281] i40e: Add functionality for FD SB to drop packets
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

With this change we can drop a flow if we wanted to.

Change-ID: I222b1ae960e61a31965bafe3159a95099e70c7d2
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 35a91fdd52cc3890764ed429d7aed816aa54f65f)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index 0a0a3b9..d2cf830 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -1513,11 +1513,16 @@ static int i40e_add_fdir_ethtool(struct i40e_vsi *vsi,
 
  input->fd_id = fsp->location;
 
+ if (fsp->ring_cookie == RX_CLS_FLOW_DISC)
+  input->dest_ctl = I40E_FILTER_PROGRAM_DESC_DEST_DROP_PACKET;
+ else
+  input->dest_ctl =
+        I40E_FILTER_PROGRAM_DESC_DEST_DIRECT_PACKET_QINDEX;
+
  input->q_index = fsp->ring_cookie;
  input->flex_off = 0;
  input->pctype = 0;
  input->dest_vsi = vsi->id;
- input->dest_ctl = I40E_FILTER_PROGRAM_DESC_DEST_DIRECT_PACKET_QINDEX;
  input->fd_status = I40E_FILTER_PROGRAM_DESC_FD_STATUS_FD_ID;
  input->cnt_index = 0;
  input->flow_type = fsp->flow_type;
-- 
1.7.1